src/library/export/dlgcoverartcopy.cpp

```diff
@@ -0,0 +1,87 @@
+#include "library/export/dlgcoverartcopy.h"
+
+#include <QMessageBox>
+
+#include "library/library_prefs.h"
+#include "util/assert.h"
+
+DlgCoverArtCopy::DlgCoverArtCopy(QWidget* parent,
+        UserSettingsPointer pConfig,
+        CoverArtCopyWorker* worker)
+        : QDialog(parent),
+          Ui::CoverArtCopyDlg(),
+          m_pConfig(pConfig),
+          m_worker(worker) {
+    setupUi(this);
+    m_coverOverwritten = false;
+
+    setModal(true);
+
+    connect(m_worker,
+            &CoverArtCopyWorker::askOverwrite,
+            this,
+            &DlgCoverArtCopy::slotAskOverwrite);
+    connect(m_worker,
+            &CoverArtCopyWorker::canceled,
+            this,
+            &DlgCoverArtCopy::btnCancelClicked);
+}
+
+void DlgCoverArtCopy::btnCancelClicked() {
+    finish();
+}
+
+void DlgCoverArtCopy::showEvent(QShowEvent* event) {
+    QDialog::showEvent(event);
+    VERIFY_OR_DEBUG_ASSERT(m_worker) {
+        qDebug() << "m_worker did not initialized, about to crash";
+        return;
+    }
+    m_worker->start();
+}
+
+void DlgCoverArtCopy::slotAskOverwrite(
+        const QString& coverArtCopyPath,
+        std::promise<CoverArtCopyWorker::OverwriteAnswer>* promise) {
+    QMessageBox overwrite_box(
+            QMessageBox::Warning,
+            tr("Overwrite Existing Cover Art?"),
+            tr("\"%1\" already exists, overwrite?").arg(coverArtCopyPath));
+    overwrite_box.addButton(QMessageBox::Yes)->setText(tr("&Overwrite"));
+    overwrite_box.addButton(QMessageBox::No)->setText(tr("&Update without Overwrite"));
```

Comment:
The user case for "Update without Overwrite" does not work on case of a MusikBrainz cover, because we have no alternative location.

Reply:
So we either overwrite or not update the cover art. Would it be okay in the use case, maybe we don't show this button for MusicBrainz and we show it for manual cover art selection.

What do you think?